src/vcpkg/commands.install.cpp

```diff
@@ -73,60 +75,99 @@ namespace vcpkg
+                if (ec)
+                {
+                    msg::println_error(msgInstallFailed, msg::path = target, msg::error_msg = ec.message());
+                }
+
+                // Trailing backslash for directories
+                this_output.push_back('/');
+                dir_output.push_back(std::move(this_output));
                 continue;
             }
 
-            const auto filename = file.filename();
-            if (vcpkg::is_regular_file(status) &&
+            const auto filename = files[i].filename();
+
+            if (is_regular_file(status) &&
                 (filename == "CONTROL" || filename == "vcpkg.json" || filename == "BUILD_INFO"))
             {
                 // Do not copy the control file or manifest file
                 continue;
             }
 
-            const auto suffix = file.generic_u8string().substr(prefix_length + 1);
-            const auto target = destination / suffix;
-
-            bool use_hard_link = true;
-            auto this_output = Strings::concat(destination_subdirectory, "/", suffix);
-            switch (status)
+            if (!is_regular_file(status) && !is_symlink(status))
             {
-                case FileType::directory:
-                {
-                    fs.create_directory(target, ec);
-                    if (ec)
-                    {
-                        msg::println_error(msgInstallFailed, msg::path = target, msg::error_msg = ec.message());
-                    }
+                msg::println_error(msgInvalidFileType, msg::path = files[i]);
+                continue;
+            }
+            files_and_status.emplace_back(std::move(files[i]), std::move(symlink_statuses[i]));
+        }
 
-                    // Trailing backslash for directories
-                    this_output.push_back('/');
-                    output.push_back(std::move(this_output));
-                    break;
-                }
-                case FileType::regular:
+        std::vector<std::string> output(files_and_status.size());
+
+        parallel_transform(
+            files_and_status.begin(), files_and_status.size(), output.begin(), [&](auto&& file_and_status) {
+                auto&& [file, maybe_status] = file_and_status;
+
+                const auto status = *maybe_status.get();
+                std::error_code ec;
+
+                const auto suffix = file.generic_u8string().substr(prefix_length + 1);
+                const auto target = destination / suffix;
+
+                bool use_hard_link = true;
+                auto this_output = Strings::concat(destination_subdirectory, "/", suffix);
+                if (is_regular_file(status))
                 {
                     if (fs.exists(target, IgnoreErrors{}))
                     {
                         msg::println_warning(msgOverwritingFile, msg::path = target);
-                        fs.remove_all(target, IgnoreErrors{});
+                        fs.remove(target, IgnoreErrors{});
```

Comment:
There's no difference for regular files between `remove_all` and `remove` (provided that vcpkg's implementation is identical to the STL): https://en.cppreference.com/w/cpp/filesystem/remove